### DISCUSSION TITLE
请问c++层面如何调用像pagView.setRepeatCount()这种设置循环次数的接口？谢谢。

### DISCUSSION THREAD

**Discussion by yinjiaoyuan** [Tencent/libpag#2534] [Q&A] [answered]
请问C++层面有像pagView.setRepeatCount()这种设置循环次数的接口吗？我想在linux/src/main.cpp中调用。谢谢。

**Comment by kevingpqi123** [comment, marked as answer]
PAGView 面向的是有 UI 框架的平台，如 iOS 的 UIView等，内部封装了 PAGPlayer、PAGSuface和 PAGFile，封装了整个播放逻辑，所以有 setRepeatCount 的接口
PAGPlayer 控制的是单帧渲染数据，你想怎么播放使用 PAGPlayer 的接口 setProgress 和 flush 就可以实现特定帧的渲染，整个播放逻辑是需要自己封装的

**Reply by yinjiaoyuan** [reply]
> PAGView 面向的是有 UI 框架的平台，如 iOS 的 UIView等，内部封装了 PAGPlayer、PAGSuface和 PAGFile，封装了整个播放逻辑，所以有 setRepeatCount 的接口 PAGPlayer 控制的是单帧渲染数据，你想怎么播放使用 PAGPlayer 的接口 setProgress 和 flush 就可以实现特定帧的渲染，整个播放逻辑是需要自己封装的

如果调用 PAGPlayer 的接口 setProgress 和 flush接口如何实现setRepeatCount（1）这种逻辑，因为好像不知道一个循环播放多少帧。我想在C++设置setRepeatCount（1）的目的是生成一个循环的图片然后复制图片就行了，因为总时长比较长的话生成整个时长的图片需要比较多的时间。

**Reply by kevingpqi123** [reply]
建议看明白 linux/src/main.cpp 这里的实现，这里实现的就是一个循环的播放，pag 文件像视频一样有帧率的概念，根据帧率和文件时长可以求出文件有多少帧，播放每一帧的时候可以转换为进度设置给 PAGPlayer
你先理解清楚帧率、时长、总帧数的概念和换算关系，这里的概念都是通用的，如视频领域

**Reply by yinjiaoyuan** [reply]
> 建议看明白 linux/src/main.cpp 这里的实现，这里实现的就是一个循环的播放，pag 文件像视频一样有帧率的概念，根据帧率和文件时长可以求出文件有多少帧，播放每一帧的时候可以转换为进度设置给 PAGPlayer 你先理解清楚帧率、时长、总帧数的概念和换算关系，这里的概念都是通用的，如视频领域

好的，我明白了，谢谢！
为你们和PAG点赞！
